### ISSUE TITLE
[release-test] long_running_many_drivers release test failed.

### ISSUE THREAD

**Issue by scv119** [ray-project/ray#35362]
### What happened + What you expected to happen

https://console.anyscale-staging.com/o/anyscale-internal/jobs/prodjob_yndvf1zbm2veplg8b7z1a2tzqc

### Versions / Dependencies

master

### Reproduction script

N/A

### Issue Severity

High: It blocks me from completing my task.

**Comment by jjyao** [comment]
```
(run_driver pid=276427) (Actor pid=279695) Inconsistency detected by ld.so: ../elf/dl-tls.c: 481: _dl_allocate_tls_init: Assertion `listp->slotinfo[cnt].gen <= GL(dl_tls_generation)' failed!
```

**Comment by can-anyscale** [comment]
@jjyao: this test passed on the latest run https://buildkite.com/ray-project/release-tests-branch/builds/1650, can you help confirm and close the issue if thing has been resolved, thanks

**Comment by cadedaniel** [comment]
@krfricke seems to have a reliable repro https://github.com/ray-project/ray/issues/32008#issuecomment-1551062307

**Comment by jjyao** [comment]
Seems it's a glibc bug: https://sourceware.org/bugzilla/show_bug.cgi?id=19329

**Comment by jjyao** [comment]
CI glibc version:

```
(base) root@6940c55dfb8c:/ray# ldd --version
ldd (Ubuntu GLIBC 2.31-0ubuntu9.9) 2.31
Copyright (C) 2020 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
Written by Roland McGrath and Ulrich Drepper.
```

The fix is in 2.34.

**Comment by jjyao** [comment]
Closing as this is a glibc bug. The only fix is upgrading glibc in our CI environment which is planned now.
